vrecord

```diff
@@ -205,6 +205,58 @@ _setup_env_variables(){
     fi
 }
 
+_get_lang(){
+    #detects system language
+    languages=( de en es fr hi id it jp nl sv vi zh )
+    language_scan1=$(locale | sed -n -e '1{p;q}')
+    for i in "${languages[@]}" ; do
+        if [[ "${language_scan1}" == *"${i}"* ]] ; then
+            system_language="${i}"
+            break
+        fi
+    done
+
+    if [[ -z "${system_language}" ]] ; then
+        language_scan2=$(locale | sed -n -e '2{p;q}' | cut -d ':' -f1)
+        for i in "${languages[@]}" ; do
+        if [[ "${language_scan2}" == *"${i}"* ]] ; then
+            system_language="${i}"
+            break
+        fi
+        done
+    fi
+    if [[ -z "${system_language}" ]] ; then system_language='en'; fi
+}
+
+_set_lang_variables(){
+    if [[ "${system_language}" == 'en' ]] ; then
+        card_var='card'
+    elif [[ "${system_language}" == 'de' ]] ; then
+        card_var='Karte'
+    elif [[ "${system_language}" == 'es' ]] ; then
+        card_var='tarjeta'
+    elif [[ "${system_language}" == 'fr' ]] ; then
+        card_var='carte'
+    elif [[ "${system_language}" == 'hi' ]] ; then
+        card_var='card'
+    elif [[ "${system_language}" == 'id' ]] ; then
+        card_var='card'
+    elif [[ "${system_language}" == 'it' ]] ; then
+        card_var='scheda'
+    elif [[ "${system_language}" == 'jp' ]] ; then
+        card_var='カード'
+    elif [[ "${system_language}" == 'nl' ]] ; then
+        card_var='kaart'
+    elif [[ "${system_language}" == 'sv' ]] ; then
+        card_var='kort'
+    elif [[ "${system_language}" == 'vi' ]] ; then
+        card_var='card'
+    elif [[ "${system_language}" == 'zh' ]] ; then
+        card_var='card'
+    fi
```

Comment:
Use a `case` statement rather than multiple `if` statements?

Reply:
I guess that might be a little bit faster, right? (Although I always find `if` easier to read/maintain than `case` for some reason - just a personal quirk I guess).

I think I'd prefer to merge this as is for now, and then maybe convert it to `case` when I circle back to fix the non-roman characters issue and and a few more languages. (Need to do a little more digging to see how many languages alsa tools supports via different words for 'card').

Reply:
LOL, I was actually thinking here about maintenance not speed.

I agree, it’a a matter of taste! I find personally this simpler (and therefore easier to maintain):

```bash
_set_lang_variables(){
    case "${system_language}" in
        en) card_var='card' ;;
        de) card_var='Karte' ;;
        es) card_var='tarjeta' ;;
        fr) card_var='carte' ;;
        hi) card_var='card' ;;
        id) card_var='card' ;;
        it) card_var='scheda' ;;
        jp) card_var='カード' ;;
        nl) card_var='kaart' ;;
        sv) card_var='kort' ;;
        vi) card_var='card' ;;
        zh) card_var='card' ;;
    esac
    _include_var card_var
}
```

Reply:
lol visually it is much simpler! I think I just hate semi-colons, both in code and in writing so my brain rebels against `case` ;-). That is much cleaner though so I'll probably take your suggestion when I revisit this. I think there are actually still quite a few localizations of alsa-tools that ideally would be  be accounted for.